sql/api/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBaseParser.g4

```diff
@@ -989,6 +989,7 @@ primaryExpression
     | CASE whenClause+ (ELSE elseExpression=expression)? END                                   #searchedCase
     | CASE value=expression whenClause+ (ELSE elseExpression=expression)? END                  #simpleCase
     | name=(CAST | TRY_CAST) LEFT_PAREN expression AS dataType RIGHT_PAREN                     #cast
+    | primaryExpression COLLATE stringLit                                                      #collate
```

Comment:
Not familiar with this part of the code, so mostly asking to understand: is the COLLATE expression expected to be applicable only to string literals? Why not string expressions in general?

Reply:
Left arg is an expression, right one is literal. E.g. you can say
`'hello' :: 'world' COLLATE 'UCS_BASIC_LCASE'.

We can make collation name to be an expression as well as needed. But it seems like the standard way to is to keep it as literal.